arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1049,7 +1049,7 @@ Result RestReplicationHandler::processRestoreCollection(VPackSlice const& collec
     return Result();
   }
 
-  ExecContextSuperuserScope escope(ExecContext::current().isAdminUser());
+  ExecContextSuperuserScope escope(ExecContext::current().isAdminUser() || !ServerState::readOnly());
```

Comment:
Should this this really be an `||`? I would have expected `&&`.

Reply:
Richtig. Ich fixe...

Reply:
LOL, I have done this correctly in 3 places but in this one I made a boolean mistake.
Thanks for catching this!